uktrade/data-hub-frontend
# Make hot module reloading (HMR) work

## Description of change

This PR makes _hot module reloading (HMR)_ work. For this to work, I had to:

- Move Webpack compilation to the Express application
- Reorganize how static assets are managed (I got rid of the file type prefixes `/js`,`/css`, `/images`, `/fonts`)
- Reorganize the two entry points (Data Hub and export win review) so that they export named constants, otherwise changes in them wouldn't be picked up by HMR
- Add a new `preview` NPM script, which allows for quickly starting the application transpiled for production
- Remove IE related code which is now obsolete
- Update functional tests testing maximum file upload size

## Test instructions

1. Update dependencies with `npm install`
2. Start the application for development with `npm run develop`
3. Load the page in your browser
4. The page should only load once Webpack middleware compiles all assets
5. You should now see Webpack `[HMR]` logs in your browser
6. Open a page of your choice in the browser and edit it's corresponding React component (e.g. `src/client/components/DashboardTabs/index.jsx` for the tab nav on the landing page)
7. Edit the react code and save
8. You should immediately see a `[HMR] bundle rebuilding` log in the browser's console
9. After the edited module is transpiled (could take a couple of seconds), you should see the change appear in the browser with a couple of more `[HMR]` logs
    ```
    [HMR] bundle rebuilding
    [HMR] bundle rebuilt in 2187ms
    [HMR] Checking for updates on the server...
    [HMR] Updated modules:
    [HMR]  - ./src/client/components/DashboardTabs/index.jsx
    [HMR] App is up to date.
    ```
10. HMR should only be enabled when `NODE_ENV=develop`
11. All this should also apply for the Export Wins Review entry point (`/exportwins/review/*`)

## Screenshots

There should be no change to the look and functionality of the application.
